kuwala/core/backend/app/controller/data_source/postgres.py

```diff
@@ -57,6 +58,29 @@ def send_query(
     return result
 
 
+def save_query(
```

Comment:
I think `write_query_result_to_file` is a better name as we are not saving the query itself but the result.